### ISSUE TITLE
tf.losses.cosine_similarity is a negative quantity between -1 and 0 

### ISSUE THREAD

**Issue by songs18** [tensorflow/tensorflow#34577]
In tensorflow website, it describes tf.losses.cosine_simialrity as follows:
Note that it is a negative quantity between -1 and 0, where 0 indicates orthogonality and values closer to -1 indicate greater similarity. 
In fact, the quantity is from 1. to -1., it just takes a negative from normal cosine_similarity.
The page is at
https://tensorflow.google.cn/api_docs/python/tf/keras/losses/cosine_similarity

**Comment by songs18** [comment]
I can find the code in github.
`
y_true = nn.l2_normalize(y_true, axis=axis)

  y_pred = nn.l2_normalize(y_pred, axis=axis)

  return -math_ops.reduce_sum(y_true * y_pred, axis=axis)
`

**Comment by gadagashwini** [comment]
@songs18, Could you post complete standalone code snippet for us to analyze the issue better?

**Comment by songs18** [comment]
‘’‘Python
def cosine_similarity(y_true, y_pred, axis=-1):
  """Computes the cosine similarity between labels and predictions.
  Note that it is a negative quantity between -1 and 0, where 0 indicates
  orthogonality and values closer to -1 indicate greater similarity. This makes
  it usable as a loss function in a setting where you try to maximize the
  proximity between predictions and targets.
  `loss = -sum(y_true * y_pred)`
  Args:
    y_true: Tensor of true targets.
    y_pred: Tensor of predicted targets.
    axis: Axis along which to determine similarity.
  Returns:
    Cosine similarity tensor.
  """
  y_true = nn.l2_normalize(y_true, axis=axis)
  y_pred = nn.l2_normalize(y_pred, axis=axis)
  return -math_ops.reduce_sum(y_true * y_pred, axis=axis)
'''

**Comment by pavithrasv** [comment]
@songs18 Thank you for bringing this up. We assume that prediction is a value between 0 and 1 and labels are binary ie. 0 or 1. With these assumptions cosine similarity will be in positive space - [0, 1] and the loss function will be in [-1, 0].

**Comment by pavithrasv** [comment]
Closing this as the behavior is as expected. Please let us know if you have any questions. Thank you!

**Comment by gitgani** [comment]
poor thing,. we cant use it for generic cosine similarity loss functions...
